conf/base.config

```diff
@@ -91,23 +91,39 @@ process {
     }
 
     // RESOURCES: MEMORY INTENSIVE STEPS, SOFTWARE TO BE UPDATED TO COMBAT THIS
-    withName: '.*:.*:SELFCOMP:(SELFCOMP_MAPIDS|SELFCOMP_MUMMER2BED|SELFCOMP_SPLITFASTA|BEDTOOLS_MERGE)' {
+    withName: '.*:.*:SELFCOMP:SELFCOMP_MUMMER2BED' {
         cpus    = { check_max( 10       * task.attempt, 'cpus'      ) }
-        memory  = { check_max( 120.GB   * task.attempt, 'memory'    ) }
+        memory  = { check_max( 1.GB     * Math.ceil( mummerfile.size() / 1e+9 )    * task.attempt, 'memory'    ) }
         time    = { check_max( 12.h     * task.attempt, 'time'      ) }
     }
 
+    withName: '.*:.*:SELFCOMP:SELFCOMP_MAPIDS' {
+        cpus    = { check_max( 10       * task.attempt, 'cpus'      ) }
+        memory  = { check_max( 1.GB     * Math.ceil( bed.size() / 1e+9 )    * task.attempt, 'memory'    ) }
+        time    = { check_max( 12.h     * task.attempt, 'time'      ) }
+    }
+
+    withName: BEDTOOLS_SORT {
+        memory  = { check_max( 1.GB     * Math.ceil( genome_file.countLines() * Math.log(10) * genome_file.countLines())        * task.attempt, 'memory'    ) }
+    }
+
     withName: SELFCOMP_ALIGNMENTBLOCKS {
         cpus    = { check_max( 20       * task.attempt, 'cpus'      ) }
-        memory  = { check_max( 120.GB   * task.attempt, 'memory'    ) }
+        memory  = { check_max( 1.GB     * (  Math.ceil( ( bedfile.size() / 1e+9 ) * 30 ) * Math.ceil( task.attempt * 1 )   ) , 'memory'  ) }
```

Comment:
isn't this related with line number in bed file?

Reply:
These are the notes I made in the meeting, 30*ref for this. Lines in bed was for BEDTOOL-SORT

Reply:
#252